f3write.c

```diff
@@ -191,21 +191,24 @@ static int create_and_fill_file(const char *path, long number, size_t size,
 	/* Create the file. */
 	full_fn = full_fn_from_number(&filename, path, number);
 	assert(full_fn);
-	if(access(full_fn, F_OK) != -1 && keep)
-	{
-		printf("Skipping file %s\n", filename);
-		return false;
-	}
+
 	printf("Creating file %s ... ", filename);
 	fflush(stdout);
-	fd = open(full_fn, O_CREAT | O_WRONLY | O_TRUNC, S_IRUSR | S_IWUSR);
+	fd = open(full_fn, O_CREAT | O_WRONLY | O_EXCL, S_IRUSR | S_IWUSR);
 	if (fd < 0) {
 		if (errno == ENOSPC) {
 			printf("No space left.\n");
 			free(full_fn);
 			return true;
 		}
-		err(errno, "Can't create file %s", full_fn);
+		if (errno == EEXIST) {
```

Comment:
Bring the line `err(errno, "Can't create file %s", full_fn);` back after the statement `if (errno == EEXIST) {...` because `errno` could be something else besides `ENOSPC` or `EEXIST`.